### PR TITLE
fleshing out some more stubs for sqlalchemy

### PR DESCRIPTION
**Pull request by herb**
used the 'try until it works' approach to filling out these stubs

### CODE REVIEW

File: third_party/2.7/sqlalchemy/engine/__init__.pyi [hunk 1]
```diff
@@ -2,5 +2,7 @@
 #
 # NOTE: This dynamically typed stub was automatically generated by stubgen.
 
+from .base import Connection, Engine, RowProxy, Transaction
```
Comment (gvanrossum) [inline third_party/2.7/sqlalchemy/engine/__init__.pyi]: If you meant those to be exported, this doesn't do that -- you'd have to say "Connection as Connection" etc.

Comment (herb) [inline third_party/2.7/sqlalchemy/engine/__init__.pyi]: interesting that `mypy` didn't complain. *shrug* forced an update with this feedback.

**Comment by gvanrossum** [comment]
There are no tests checking that the stubs actually match the runtime API. When you said "try until it works" I presumed you actually tested that your app type-checks with these. I now understand that you just ran `runtests.py`?

**Comment by herb** [comment]
No, your original presumption was correct--I tested against my app until type checking didn't complain. Specifically, I installed `mypy` in a virtualenv. Then modified `typeshed` in the virtualenv and ran `mypy`. Reepat until no error. Here's a toy reproduction on a week old version of `mypy`:

```
(venv) $ cat test.py
from sqlalchemy.engine import Connection

def test(conn):
    # type: (Connection) -> None
    assert conn
(venv) $ mypy --py2 test.py
test.py:1: error: Module has no attribute 'Connection'
(venv) $ echo "from .base import Connection" >> venv/lib/mypy/typeshed/third_party/2.7/sqlalchemy/engine/__init__.pyi
(venv) $ echo "class Connection(object): ..." >> venv/lib/mypy/typeshed/third_party/2.7/sqlalchemy/engine/base.pyi
(venv) $ mypy --py2 test.py
(venv) $
```

Is this a invalid way of verifying my changes? Happy to try something else to increase confidence here.

**Comment by gvanrossum** [comment]
That sounds like a fine testing approach. I'm not sure exactly where I went wrong, maybe mypy still lets you do an explicit import of a non-public attribute, or maybe __init__ is special. Merging.

**Comment by gvanrossum** [comment]
Oh, without the 'as ...' phrasing, 'import *' doesn't work.

**Pull request merged**